### ISSUE TITLE
Failed to load Starlark extension '@com_github_mjbots_bazel_deps//tools/workspace:makefile.bzl'. Cycle in the workspace file detected

### ISSUE THREAD

**Issue by terrisgit** [mjbots/bazel_deps#12]
Please see https://github.com/dev-guy/bazel-deps-example

**Comment by terrisgit** [comment]
This is how my WORKSPACE starts:

# bazel_deps (begin)
load("//tools/workspace:default.bzl", "add_default_repositories")

add_default_repositories()

load(
    "@bazel_deps//tools/workspace:default.bzl",
    bazel_deps_add = "add_default_repositories",
)

bazel_deps_add()

**Comment by terrisgit** [comment]
After tweaking some things, now I'm getting:

```
ERROR: Failed to load Starlark extension '@com_github_mjbots_bazel_deps//tools/workspace/ffmpeg:config.bzl'.
Cycle in the workspace file detected. This indicates that a repository is used prior to being defined.
The following chain of repository dependencies lead to the missing definition.
 - @com_github_mjbots_bazel_deps
This could either mean you have to add the '@com_github_mjbots_bazel_deps' repository with a statement like `http_archive` in your WORKSPACE file (note that transitive dependencies are not added automatically), or move an existing definition earlier in your WORKSPACE file.
```

**Comment by jpieper** [comment]
Could you please post a branch to some repo that reproduces the failure?

**Comment by terrisgit** [comment]
I'm using master. Is there a way to turn off ffmpeg:config.bzl ?

**Comment by terrisgit** [comment]
```
def bazel_deps_repository(name):
    commit = "master"
    http_archive(
        name = name,
        url = "https://github.com/mjbots/bazel_deps/archive/{}.zip".format(commit),
        # sha256 = "XXX",
        strip_prefix = "bazel_deps-{}".format(commit),
    )
```

**Comment by jpieper** [comment]
re ffmpeg, it is probably best to keep github issues single topic.  That said, you can set `excludes` when you call `@bazel_deps//tools/workspace/default.bzl:add_default_repositories` to exclude any subset of packages.

Regarding the latest paste, in order to debug your initial problem, I was hoping you could push a branch I could pull with git to repro the problem.

**Comment by terrisgit** [comment]
I understand but I was just following your readme.

**Comment by terrisgit** [comment]
Any project that follows your README and has the above in the WORKSPACE file fails to build. Can you please point me to a sample repo that uses bazel_deps?

**Comment by terrisgit** [comment]
Please see https://github.com/dev-guy/bazel-deps-example

**Comment by jpieper** [comment]
That error is just bazel's terrible way of saying that it doesn't know how to resolve a particular external repository name.  Here, it is because some things in the bazel_deps repo expect its name to be `com_github_mjbots_bazel_deps` but the README was just using `bazel_deps`.

I proposed a fix in #14, since apparently the name is not really configurable anymore.  How does that work for you?

**Comment by jpieper** [comment]
#14 merged
